### PR TITLE
.github/ISSUE_TEMPLATE: add bug report and package request issue forms

### PR DESCRIPTION
**Pull request by classabbyamp**
Co-authored-by: @0x5c 

This PR replaces the existing markdown-based issue template with two new [Github issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms), one for bug reports and one for package requests.

These ensure that the relevant information is asked for and the required information is provided. They also automagically tag issues with the `bug` and `request` tags. If preferred, new bug reports could be also be marked with a second `unconfirmed` or `needs triage` label, applied by the form.

Issues that do not fall under these types (like RFCs) can still be opened by clicking the "open a blank issue" button on the template chooser:
image
This can be limited to maintainers by changing `blank_issues_enabled` to `false` in `.github/ISSUE_TEMPLATE/config.yml` if desired.

`config.yml` can also be used to add additional arbitrary links in the template chooser. Here's an example [config.yml](https://github.com/miaowware/qrm2/blob/master/.github/ISSUE_TEMPLATE/config.yml) and resulting [template chooser](https://github.com/miaowware/qrm2/issues/new/choose) in another project.

#### Testing the changes
- I tested the changes in this PR: **YES**
    A preview of the rendered templates can be seen [here](https://github.com/classabbyamp/void-packages-test/issues/new/choose)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/pkg-request.yml
Comment (paper42) [inline .github/ISSUE_TEMPLATE/pkg-request.yml:31]: this is an issue template, so "start a PR" is not relevant

File: .github/ISSUE_TEMPLATE/pkg-request.yml
Comment (paper42) [inline .github/ISSUE_TEMPLATE/pkg-request.yml:32]:
```suggestion
        while not meeting any of the following requirements, is a good candidate for inclusion in Void packages.
```
This sounds better to me, but I am not a native speaker, what do you think?

**Review by paper42 — COMMENTED** [review]
I think it would be nice to have a checkbox for electron applications that would add the electron tag and maybe an option (and a new tag) for proprietary software.

**Comment by paper42** [comment]
I don't like that we have a lot of static text in the request issue template. The whole "Quality Requirements" section could be replaced with a link to Contributing.md.

**Comment by classabbyamp** [comment]
> I think it would be nice to have a checkbox for electron applications that would add the electron tag and maybe an option (and a new tag) for proprietary software.

Those could definitely be included as checkboxes, but it's beyond the scope of issue forms to dynamically add tags like that. The new [issue automation](https://docs.github.com/en/issues/trying-out-the-new-projects-experience/automating-projects) can probably do such a thing, but that would require a much larger change (migrating to the new issues/projects entirely). Using a bot that looks for the checkbox/text is another option, which could fit in well with the bot @0x5c has been hacking at that tries to autotag PRs based on changeset contents etc.

> I don't like that we have a lot of static text in the request issue template. The whole "Quality Requirements" section could be replaced with a link to Contributing.md.

Should've been a bit more clear there, the explanatory text does not get included in the final issue, it's just for the person filling out the form to read while the form is being filled.

File: .github/ISSUE_TEMPLATE/pkg-request.yml
Comment (0x5c) [reply .github/ISSUE_TEMPLATE/pkg-request.yml:32]: Your suggestion sounds better to me

**Comment by 0x5c** [comment]
> I think it would be nice to have a checkbox for electron applications that would add the electron tag and maybe an option (and a new tag) for proprietary software.

As Abby said, that's not directly possible. However, using a bot that looks for a specific variant of a magic string seems prone to breakage.

**Comment by 0x5c** [comment]
> * checkboxes re-introduce _tasks_ we have just got rid of

This is behaviour we did not expect, as it was different at some point in the past (simply using text instead of markdown checkboxes). We'll do some tests to see what can be done about that, y/n dropdowns are likely to do the trick

> * mandatory checkboxes are dumb on their own anyway

The goal is to limit what can be requested (only released software, which is void's policy on new packages) and limit duplicates reports (which are extra work and are likely to create obsolete reports)

> * in bug report: there is no need for _update of package_ section

You're likely to have someone who considers old, but otherwise functioning, versions of packages to be "bugs". That text can be removed but it's not gonna be a meaningful change since it's not rendered in the final issue anyways.

> * in bug report: system info should not be mandatory, sometimes it's obvious it doesn't matter

I wonder if it's more obvious to maintainers than to bug reporters, and I suspect that without it being provided, the case it's needed would only be discovered beyond the point the reporter has gone AWOL

> * in bug report: _Bug description_ section should keep it's current name, _Actual behavior_
> * in package request: there should be a field to provide homepage

Consider those fixed

**Comment by classabbyamp** [comment]
Updated to address most of the issues raised, I think. @Chocimier if you want to update your live version, go ahead. The links in the original comment should still work for previewing.

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (paper42) [inline .github/ISSUE_TEMPLATE/bug-report.yml:24]: Could we mention that people can get these strings with `xbps-query -p pkgver foo`?

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (paper42) [inline .github/ISSUE_TEMPLATE/bug-report.yml:3]: I am not sure I like labeling an issue as a bug when a bug report is filed. Now the "bug" label is assigned manually by a maintainer and I assign them only when I know the issue isn't a user error and someone was able to reproduce it. That said, I can see the other point of view - all bug reports should be labeled as such.

**Comment by classabbyamp** [comment]
> free-form issue is too hard to find, consider making it a third template

We've found a way to make this work without adding a template:

image

A freeform/blank _template_ is not possible when using issue forms, and a freeform issue form is not really possible (it would have to be a single multi-line input and would add a weird title to the body of the issue). This solution also makes the button to create a blank issue less prominent than the bright green ones for bugs and requests.

The latest updates are available for live preview here: https://github.com/classabbyamp/void-packages-test/issues/new/choose
(note that the current link for freeform issues is pointed to the actual void-packages repo, as it will be after merging. It also currently shows the old issue template since this PR hasn't been merged)

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (classabbyamp) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: addressed

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (0x5c) [reply .github/ISSUE_TEMPLATE/bug-report.yml:3]: The form can apply more labels, which could be `unconfirmed` or `needs-triage`, to be removed by a maintainer when confirmed.

Another approach is keeping the automatic labels as-is, but adding to the repo the usage of a `confirmed` label, to be added by a maintainer when the bug in confirmed.  
This might be better if there's interest in excluding confirmed bugs from stalebot.

The idea of both approaches is to ensure that anything claimed to be a bug can actually be found without fumbling with github's horrible keyword search. Searching by labels is more capable and allows searching for combinations, ie: `label:bug -label:unconfirmed` to only find confirmed bugs.

**Comment by Chocimier** [comment]
Remarks from paper42 to replace quality requirement with link and to not tag are right. Otherwise quite fine for me.

**Comment by 0x5c** [comment]
> Remarks from paper42 to replace quality requirement with link and to not tag are right. Otherwise quite fine for me.

That text had to be edited to make it more comprehensible to package requesters, since the manual's wording is heavily intended for those who are writing a template, and mentions things like `If you believe you have an exception, start a PR and make an argument for why that particular piece of software, while not meeting any of the following requirements, is a good candidate for the Void packages system.`, which seems fairly nonconstructive in the context of a package request.

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (paper42) [inline .github/ISSUE_TEMPLATE/bug-report.yml:61]: image
Newlines are missing in a webkit based browser, but I don't think we can fix that.

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (paper42) [reply .github/ISSUE_TEMPLATE/bug-report.yml:3]: I like the idea of an "unconfirmed" label :+1:

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (classabbyamp) [reply .github/ISSUE_TEMPLATE/bug-report.yml:61]: works fine in chromium-based, but yeah I think you're right

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (classabbyamp) [reply .github/ISSUE_TEMPLATE/bug-report.yml:3]: if you create the label, we can add it to the PR

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (paper42) [reply .github/ISSUE_TEMPLATE/bug-report.yml:61]: This is from Epiphany 42 (GNOME Web).

Comment (classabbyamp) [reply .github/ISSUE_TEMPLATE/bug-report.yml:61]: hm, sounds like something that should be reported [upstream](https://github.com/github/feedback/discussions/categories/issues-feedback)

**Comment by Chocimier** [comment]
> edited to make it more comprehensible

I see no meaningful rewording, but if any is needed, please send PR changing proper documentation. Maintaining many copies of same text is not sustainable.

**Comment by classabbyamp** [comment]
Updated.

image

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (0x5c) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: Any news on this?

Comment (classabbyamp) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: added the new `needs-testing` label to the bug report template by default, which should solve this.

**Comment by classabbyamp** [comment]
latest version of this is [live here](https://github.com/classabbyamp/void-packages-test/issues/new/choose). Any critiques?

**Review by paper42 — APPROVED** [review]

**Review by Chocimier — APPROVED** [review]

**Comment by Chocimier** [comment]
"is this new" checkbox should go first to avoid wasting reporter's time

**Review by jcgruenhage — APPROVED** [review]

**Pull request merged**
